pkg/cmd/pr/create/create.go

```diff
@@ -250,154 +483,121 @@ func createRun(opts *CreateOptions) error {
-		}
-
-		if isTerminal {
-			fmt.Fprintf(opts.IO.ErrOut, message,
-				cs.Cyan(headBranchLabel),
-				cs.Cyan(baseBranch),
-				ghrepo.FullName(baseRepo))
-			if (title == "" || body == "") && defaultsErr != nil {
-				fmt.Fprintf(opts.IO.ErrOut, "%s warning: could not compute title or body defaults: %s\n", cs.Yellow("!"), defaultsErr)
-			}
-		}
+	params := map[string]interface{}{
+		"title":       state.Title,
+		"body":        state.Body,
+		"draft":       opts.IsDraft,
+		"baseRefName": ctx.BaseBranch,
+		"headRefName": ctx.HeadBranchLabel,
 	}
 
-	tb := shared.IssueMetadataState{
-		Type:       shared.PRMetadata,
-		Reviewers:  opts.Reviewers,
-		Assignees:  opts.Assignees,
-		Labels:     opts.Labels,
-		Projects:   opts.Projects,
-		Milestones: milestoneTitles,
+	if params["title"] == "" {
+		return errors.New("pull request title must not be blank")
 	}
 
-	if !opts.WebMode && !opts.Autofill && opts.Interactive {
-		var nonLegacyTemplateFiles []string
-		var legacyTemplateFile *string
-
-		if opts.RootDirOverride != "" {
-			nonLegacyTemplateFiles = githubtemplate.FindNonLegacy(opts.RootDirOverride, "PULL_REQUEST_TEMPLATE")
-			legacyTemplateFile = githubtemplate.FindLegacy(opts.RootDirOverride, "PULL_REQUEST_TEMPLATE")
-		} else if rootDir, err := git.ToplevelDir(); err == nil {
-			nonLegacyTemplateFiles = githubtemplate.FindNonLegacy(rootDir, "PULL_REQUEST_TEMPLATE")
-			legacyTemplateFile = githubtemplate.FindLegacy(rootDir, "PULL_REQUEST_TEMPLATE")
-		}
+	err = shared.AddMetadataToIssueParams(client, ctx.BaseRepo, params, &state)
+	if err != nil {
+		return err
+	}
 
-		editorCommand, err := cmdutil.DetermineEditor(opts.Config)
-		if err != nil {
-			return err
+	pr, err := api.CreatePullRequest(client, ctx.BaseRepo.(*api.Repository), params)
+	if pr != nil {
+		fmt.Fprintln(opts.IO.Out, pr.URL)
+	}
+	if err != nil {
+		if pr != nil {
+			return fmt.Errorf("pull request update failed: %w", err)
 		}
+		return fmt.Errorf("pull request create failed: %w", err)
+	}
+	return nil
+}
 
-		err = shared.TitleBodySurvey(opts.IO, editorCommand, &tb, client, baseRepo, title, body, defs, nonLegacyTemplateFiles, legacyTemplateFile, true, baseRepo.ViewerCanTriage())
-		if err != nil {
-			return fmt.Errorf("could not collect title and/or body: %w", err)
-		}
+func previewPR(opts CreateOptions, ctx CreateContext, state shared.IssueMetadataState) error {
+	openURL, err := generateCompareURL(ctx, state)
+	if err != nil {
+		return err
+	}
 
-		action = tb.Action
+	if opts.IO.IsStdinTTY() && opts.IO.IsStdoutTTY() {
+		fmt.Fprintf(opts.IO.ErrOut, "Opening %s in your browser.\n", utils.DisplayURL(openURL))
+	}
+	return utils.OpenInBrowser(openURL)
 
-		if action == shared.CancelAction {
-			fmt.Fprintln(opts.IO.ErrOut, "Discarding.")
-			return nil
-		}
+}
 
-		if title == "" {
-			title = tb.Title
-		}
-		if body == "" {
-			body = tb.Body
-		}
-	}
+func handlePush(opts CreateOptions, ctx CreateContext) error {
```

Comment:
I also found this name a bit confusing mainly because it didn't have the word `branch` in it. I would say `pushBranch` would be good but we actually use that name inside this function, so maybe `conditionallyPushBranch`?

Reply:
I think because of the amount of stuff that function does any name is going to be inaccurate and warrant human review; I'm ok with the ambiguity of "handle" for now.